File: src/main/java/homeWorkSix/service/ProductServiceImpl.java
package homeWorkSix.service;

import homeWorkSix.model.Product;
import homeWorkSix.model.Water;

import java.io.FileReader;
import java.io.FileWriter;
import java.util.logging.Logger;

public class ProductServiceImpl implements ProductService{
    Logger logger= Logger.getAnonymousLogger();

    @Override
    public void saveProduct(Product product) {
    }

    @Override
    public void saveProduct(Water water) {
        try(FileWriter fileWriter = new FileWriter("water.txt")){
            fileWriter.write(water.toString());
        }catch (Exception e){
            logger.warning("Houston, we have a problem");
        }
    }

    @Override
    public void getProduct() {
        try(FileReader fileReader = new FileReader("water.txt")){
            int c;
            while ((c=fileReader.read())!=-1){
                System.out.print((char) c);
            }
        }catch (Exception e){
            logger.warning("Houston, we have a problem");
        }
    }
}

File: src/main/java/homeWorkSeven/service/ProductServiceImpl.cs
using service;

namespace service
{
    public class ProductServiceImpl : ProductService 
    {
    
    Logger logger= Logger.getAnonymousLogger();

    
        public void SaveProduct(Product product) {}

        
        public void SaveProduct(Water water) { 
            try
            {
                StreamWriter sw = new StreamWriter("water.txt");
                sw.WriteLine(water.toString());
                sw.Close();
            }
            catch(Exception e)
            {
                logger.warning("Houston, we have a problem");
            }
            finally
            {
                Console.WriteLine("Executing finally block.");
            }
            }

        public void GetProduct() {
            String line;
            try
            {
                StreamReader sr = new StreamReader("water.txt");
                line = sr.ReadLine();
                while (line != null)
                {
                    Console.WriteLine(line);
                    line = sr.ReadLine();
                }
                sr.Close();
                Console.ReadLine();
            }
            catch(Exception e)
            {
                logger.warning("Houston, we have a problem");
            }
            finally
            {
                Console.WriteLine("Executing finally block.");
            }

            
        }
    }
}
